blueocean-dashboard/src/main/js/creation/bitbucket/server/steps/BbChooseServerStep.jsx

```diff
@@ -0,0 +1,104 @@
+import { observer } from 'mobx-react';
+import { Dropdown, FormElement } from '@jenkins-cd/design-language';
+
+import FlowStep from '../../../flow2/FlowStep';
+import BbAddServerDialog from '../BbAddServerDialog';
+
+
+let t = null;
+
+
+@observer
+class BbChooseServerStep extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedServer: null,
+            showAddServerDialog: false,
+        };
+
+        this.dropdown = null;
+    }
+
+    componentWillMount() {
+        t = this.props.flowManager.translate;
+    }
+
+    _onChangeServerDropdown(option) {
+        this.setState({
+            selectedServer: option,
+        });
+    }
+
+    _onClickAddButton() {
+        this.setState({
+            showAddServerDialog: true,
+        });
+    }
+
+    _onAddServerDialogClosed(newServer) {
+        const newState = {
+            showAddServerDialog: false,
+        };
+
+        if (newServer) {
+            newState.selectedServer = newServer;
+        }
+
+        this.setState(newState);
+
+        // TODO: control this more cleanly via a future 'selectedOption' prop on Dropdown
+        this.dropdown.setState({
+            selectedOption: newServer,
+        });
+    }
+
+    _onClickNextButton() {
+        this.props.flowManager.selectServer(this.state.selectedServer);
+    }
+
+    render() {
+        const { flowManager } = this.props;
+        const { serverManager } = flowManager;
+        const title = t('creation.bbserver.choose_server.title');
+        const disabled = flowManager.stepsDisabled;
+        const disabledNext = !this.state.selectedServer;
+
+        return (
+            <FlowStep {...this.props} className="github-enterprise-choose-server-step" disabled={disabled} title={title}>
+                <FormElement title={t('creation.bbserver.choose_server.instructions')}>
+                    <Dropdown
+                        ref={dropdown => { this.dropdown = dropdown; }}
+                        className="dropdown-server"
+                        options={serverManager.servers}
+                        labelField="apiUrl"
+                        onChange={option => this._onChangeServerDropdown(option)}
+                    />
+                    <button className="button-add-server btn-secondary" onClick={() => this._onClickAddButton()}>
+                        {t('creation.bbserver.choose_server.button_add')}
+                    </button>
+                </FormElement>
+
+                { this.state.showAddServerDialog &&
+                <BbAddServerDialog
+                    flowManager={flowManager}
+                    onClose={cred => this._onAddServerDialogClosed(cred)}
+                />
+                }
+
+                <button className="button-next-step" disabled={disabledNext} onClick={() => this._onClickNextButton()}>
+                    {t('creation.githubent.choose_server.button_next')}
```

Comment:
Yeah, that should be fine. I am going to address these i18n comments, I will see if I can do it myself otherwise separate ticket for doing this is fine as well.